cocos/base/CCUserDefault.cpp

```diff
@@ -332,6 +332,10 @@ Data UserDefault::getDataForKey(const char* pKey, const Data& defaultValue)
             ret.fastSet(decodedData, decodedDataLen);
```

Comment:
Could you check other places where fastSet is used?
I think that this method should be removed.

Reply:
`CCData` has an unintuitive interface and incomplete documentation. I hope it behaves like smart pointers, since it manages memory ownership.

`fastSet` could be like `unique_ptr::reset(p)` by freeing previously managed memory, and obtaining ownership of `p`.

`copy` method has somewhat misleading documentation.

https://github.com/cocos2d/cocos2d-x/blob/4afda3a9abec62fa111396c19ea30cc31c11ea85/cocos/base/CCData.h#L96-L102

> Developer should free the pointer after invoking this method. 

Which pointer does it refers to in `free the pointer`? If it means the `buffer pointer`, we could call `reset` instead?